File: mvp/src/main/java/com/quyunshuo/mvp/view/MainActivity.java
package com.quyunshuo.mvp.view;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.quyunshuo.mvp.MainContract;
import com.quyunshuo.mvp.R;
import com.quyunshuo.mvp.presenter.WeatherPresenter;

/**
 * Activity作为View层 实现View层接口
 */
public class MainActivity extends AppCompatActivity implements MainContract.IWeatherView {

    WeatherPresenter mPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public void onInfoUpdate(String info) {

    }

    @Override
    public void showWaitingDialog() {

    }

    @Override
    public void closedWaitingDialog() {

    }
}

File: circularreveal/src/main/java/com/example/circularreveal/MainActivity.kt
package com.example.circularreveal

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        main_hello.setOnClickListener { view ->
            val intent = Intent(this, SecondActivity::class.java)
            val location = IntArray(2)
            view.getLocationInWindow(location)
            //把点击按钮的中心位置坐标传过去作为 SecondActivity 的揭露动画圆心
            intent.putExtra(CLICK_X, location[0] + view.width / 2)
            intent.putExtra(CLICK_Y, location[1] + view.height / 2)
            startActivity(intent)
        }
    }
}
